src/index.js

```diff
@@ -87,19 +168,35 @@ function getTestMetadataDeclaration(state, t) {
   ]);
 }
 
-function shouldLoadFile(filename) {
-  return filename.match(/[-_]test\.js/gi);
-}
-
-function hasMetadataDeclaration({ node }) {
-  if (node.expression && node.expression.type === 'AssignmentExpression') {
-    return (
-      node.expression.left.object.name === 'testMetadata' &&
-      node.expression.left.property.name === 'filePath'
-    );
-  } else {
+/**
+ * Checks whether a test metadata statement already exists.
+ * @param {object} node - Babel node path
+ * @param {object} t - Babel types
+ * @returns {Boolean}
+ */
+function hasMetadataDeclaration({ node }, t) {
+  if (
+    !node.expression ||
+    !t.isAssignmentExpression(node.expression) ||
+    !node.expression.left.object ||
+    !node.expression.left.property
+  ) {
     return false;
   }
```

Comment:
I may be inclined to store this in a variable, and && it off the boolean below, in which case it'd short circuit if `false`.